lib/executor/common_test.go

```diff
@@ -13,23 +12,29 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func setupExecutor(
-	t *testing.T, config lib.ExecutorConfig,
-	vuFn func(context.Context, chan<- stats.SampleContainer) error,
-) (context.Context, context.CancelFunc, lib.Executor, *testutils.SimpleLogrusHook) {
+func simpleRunner(vuFn func(context.Context) error) lib.Runner {
+	return &testutils.MiniRunner{
+		Fn: func(ctx context.Context, _ chan<- stats.SampleContainer) error {
+			return vuFn(ctx)
```

Comment:
The `stats.SampleContainer` channel is unused, which is why you're not passing it anymore, right?

Reply:
yes, because none of the simple functions we've used in the tests have required the output - that's why I called it `simpleRunner`. But I changed the `setupExecutor` to expect the `lib.Runner` interface, so that you can always pass very complicated `MiniRunner` instances, if you need to.